tests/test_pagure.py

```diff
@@ -0,0 +1,749 @@
+
+    @httpretty.activate
+    def test_get_from_date_issues(self):
+        """Test issues from date API call"""
+
+        issues = read_file('data/pagure/pagure_repo_issue_from_2020_03_07')
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_ISSUES_URL,
+                               body=issues,
+                               status=200,
+                               )
+
+        from_date = datetime.datetime(2020, 3, 7)
+        client = PagureClient(namespace=None, repository='Project-example', token='aaa')
+
+        raw_issues = [issues for issues in client.issues(from_date=from_date)]
+        self.assertEqual(raw_issues[0], issues)
+
+        # Check requests
+        expected = {
+            'status': ['all'],
+            'per_page': ['100'],
+            'order': ['asc'],
+            'since': ['2020-03-07 00:00:00']
+        }
+        self.assertDictEqual(httpretty.last_request().querystring, expected)
+        self.assertEqual(httpretty.last_request().headers["Authorization"], "token aaa")
+
+    @httpretty.activate
+    def test_get_empty_issues(self):
+        """ Test when issue is empty API call """
+
+        issue = read_file('data/pagure/pagure_empty_request')
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_ISSUES_URL,
+                               body=issue, status=200,
+                               )
+
+        client = PagureClient(namespace=None, repository="Project-example", token="aaa")
+
+        raw_issues = [issues for issues in client.issues()]
+        self.assertEqual(raw_issues[0], issue)
+
+        # Check requests
+        expected = {
+            'status': ['all'],
+            'per_page': ['100'],
+            'order': ['asc']
+        }
+
+        self.assertDictEqual(httpretty.last_request().querystring, expected)
+        self.assertEqual(httpretty.last_request().headers["Authorization"], "token aaa")
+
+    @httpretty.activate
+    def test_http_wrong_status(self):
+        """Test if a error is raised when the http status was not 200"""
+
+        issue = ""
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_ISSUES_URL,
+                               body=issue,
+                               status=501,
+                               )
+
+        client = PagureClient(namespace=None, repository="Project-example", token="aaa", sleep_time=1, max_retries=1)
+
+        with self.assertRaises(requests.exceptions.HTTPError):
+            _ = [issues for issues in client.issues()]
+
+        # Check requests
+        expected = {
+            'status': ['all'],
+            'per_page': ['100'],
+            'order': ['asc']
+        }
+
+        self.assertDictEqual(httpretty.last_request().querystring, expected)
+        self.assertEqual(httpretty.last_request().headers["Authorization"], "token aaa")
+
+    @httpretty.activate
+    def test_http_retry_error(self):
```

Comment:
This test isn't needed anymore if the extra_status_forcelist is removed from the pagureclient